plugin.video.carnet-meduza/resources/lib/carnet-meduza.py

```diff
@@ -0,0 +1,264 @@
+api_base_url = 'https://meduza.carnet.hr/index.php/api/'
+category_image_base_url = 'https://meduza.carnet.hr/uploads/images/categories/'
+# api_key == device_id from register (import) device == uid
+api_key = addon.getSetting('apikey')
+
+mode = args.get('mode', None)
+
+dir_recommends = addon.getLocalizedString(30201)
+dir_categories = addon.getLocalizedString(30202)
+dir_channels = addon.getLocalizedString(30203)
+dir_search = addon.getLocalizedString(30204)
+dir_settings = addon.getLocalizedString(30205)
+
+# get api lang property (en|hr)
+active_lang = xbmc.getLanguage()
+if active_lang == 'Croatian':
+	lang_api_prop = 'naziv'
+else:
+	lang_api_prop = 'naziv_en'
+
+if mode is None:
+    url = build_url({'mode': 'folder', 'foldername': dir_recommends})
+    li = xbmcgui.ListItem(dir_recommends, iconImage='DefaultVideo.png')
+    xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li, isFolder=True)
+
+    url = build_url({'mode': 'folder', 'foldername': dir_categories})
+    li = xbmcgui.ListItem(dir_categories, iconImage='DefaultAddonVideo.png')
+    xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li, isFolder=True)
+
+    url = build_url({'mode': 'folder', 'foldername': dir_channels})
+    li = xbmcgui.ListItem(dir_channels, iconImage='DefaultAddonTvInfo.png')
+    xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li, isFolder=True)
+
+    url = build_url({'mode': dir_search, 'foldername': dir_search})
+    li = xbmcgui.ListItem(dir_search, iconImage='DefaultAddonsSearch.png')
+    xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li, isFolder=True)
+	
+    url = build_url({'mode': dir_settings, 'foldername': dir_settings})
+    li = xbmcgui.ListItem(dir_settings, iconImage='DefaultAddonProgram.png')
+    xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li)
+	
+    xbmcplugin.endOfDirectory(addon_handle)
+
+elif mode[0] == 'folder':
+	if args['foldername'][0].decode('utf-8') == dir_recommends:
+		videos =  recommended_videos()
+		list_search_or_recommended_videos(videos)
+
+	elif args['foldername'][0].decode('utf-8') == dir_categories:
+		categories = categories()
+		categoryGen = (category for category in categories if category['naziv'] != 'YouTube')
+		for category in categoryGen:
+			active_lang = xbmc.getLanguage()
+			name = category[lang_api_prop].encode('utf-8')
+			url = build_url({'mode': 'folder', 'foldername': name})
+			categoryImage = category_image_base_url + category['slika']
+			li = xbmcgui.ListItem(name, iconImage=categoryImage)
+			xbmcplugin.addDirectoryItem(handle=addon_handle, url=url, listitem=li, isFolder=True)
+		xbmcplugin.endOfDirectory(addon_handle)
+
+	elif args['foldername'][0].decode('utf-8') == dir_channels:
+		channels = channels()
+		for channel in channels:
+			name = channel['naziv'].encode('utf-8')
+			url = build_url({'mode': 'folder', 'foldername': name})
+			try:
+				channelImage = channel['slika']
```

Comment:
Rather than using try - except, you can use:
 channelImage = channel.get('slika','')

which is much simplier and more efficient than the try-except